android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -579,6 +640,74 @@ private void showVpnCalloutDialog() {
+        if (findViewById(R.id.news_settings_bar) != null) {
+            return;
+        }
+        if (compositorView != null && controlContainer != null) {
+            int[] coords = {0, 0};
+            controlContainer.getLocationOnScreen(coords);
+            int absoluteTop = coords[1];
+            int absoluteBottom = coords[1] + controlContainer.getHeight();
+
+            LayoutInflater inflater = LayoutInflater.from(this);
+            // inflate the settings bar layout
+            View inflatedSettingsBarLayout =
+                    inflater.inflate(R.layout.brave_news_settings_bar_layout, null);
+            RelativeLayout newContentButtonLayout =
+                    (RelativeLayout) inflater.inflate(R.layout.brave_news_load_new_content, null);
+            // add the bar to the layout stack
+            compositorView.addView(inflatedSettingsBarLayout, 2);
+            compositorView.addView(newContentButtonLayout, 3);
+            inflatedSettingsBarLayout.setAlpha(0f);
+            FrameLayout.LayoutParams inflatedLayoutParams =
+                    new FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, 100);
+
+            FrameLayout.LayoutParams newContentButtonLayoutParams = new FrameLayout.LayoutParams(
+                    FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT);
+
+            // position bellow the control_container element (nevigation bar) with 25dp compensation
+            inflatedLayoutParams.setMargins(0, controlContainer.getBottom() - 5, 0, 0);
+            newContentButtonLayoutParams.setMargins(0, 400, 0, 0);
+            newContentButtonLayoutParams.gravity = Gravity.CENTER_HORIZONTAL;
+            newContentButtonLayout.setGravity(Gravity.CENTER_HORIZONTAL);
+            inflatedSettingsBarLayout.setLayoutParams(inflatedLayoutParams);
+            newContentButtonLayout.setLayoutParams(newContentButtonLayoutParams);
+
+            inflatedSettingsBarLayout.setVisibility(View.VISIBLE);
+
+            compositorView.invalidate();
+        }
+    }
+
+    public void removeSettingsBar() {
+        CompositorViewHolder compositorView = findViewById(R.id.compositor_view_holder);
+
+        if (compositorView != null) {
+            View settingsBar = compositorView.getChildAt(2);
+            if (settingsBar != null) {
+                if (settingsBar.getId() == R.id.news_settings_bar) {
+                    compositorView.removeView(settingsBar);
+                }
+            }
+        }
+    }
+
+    // Sets NTP background
+    public void setBackground(BitmapDrawable bgWallpaper) {
+        CompositorViewHolder compositorView = findViewById(R.id.compositor_view_holder);
+
+        ViewGroup root = (ViewGroup) compositorView.getChildAt(1);
+        ScrollView scrollView = (ScrollView) root.getChildAt(0);
+        scrollView.setId(View.generateViewId());
+
+        //@TODO alex use Glide to set the image?
```

Comment:
? Should this be fixed before merge?

Reply:
it can be done after, since there will be other things to do anyway. this change would be an improvement with the introduction of Glide library.